plugin.video.mytv_bg/addon.py

```diff
@@ -168,50 +245,95 @@ def tvList(url):
 			for (key, val) in enumerate(menulist):
 
 				if val['type'] == 'item':
+					url_parameters = {
+					'tracking': getTrackingKey(val),
+					'title' : ''
+					}
+
 					items.append({
 						'label': u"{0}".format(val['title']),
+						'label2': u"{0}".format(getTrackingKey(val)),
 						'key': u"{0}".format(key),
-						'url': plugin.url_for('tvPlay', url=val['key'], tracking_key=getTrackingKey(val)),
+						'url': router.url_for('tvPlayCaller', urllib.quote_plus(val['key'])+'?'+urllib.urlencode(url_parameters)),
 						'thumb': "{0}".format(val["thumb"])})
 				elif val['type'] == 'menu':
 					label = val['title'].encode('utf-8')
+
+					url_parameters = {
+					'show_title': _show_title,
+					'title' : label
+					}
 					items.append({
 						'label': label,
 						'key': u"{0}".format(key),
-						'url': plugin.url_for('tvList', url=val['key'], title=label, show_title=_show_title),
+						'url': router.url_for('tvList', urllib.quote_plus(val['key'])+'?'+urllib.urlencode(url_parameters)),
 						'thumb': "{0}".format(val["thumb"])})
 
 
 
 	return plugin.add_items(items, False, [xbmcplugin.SORT_METHOD_VIDEO_SORT_TITLE, xbmcplugin.SORT_METHOD_VIDEO_TITLE])
 
-def getTrackingKey(val):
-	if 'tracking_key' in val:
-		return val['tracking_key']
-	return None
 
-@plugin.route('/tvPlay/<url>')
+
+@router.route('/tvPlayCaller/<url>')
+def tvPlayCaller(url):
+	args = urlparse.parse_qs(sys.argv[2][1:], keep_blank_values=True, strict_parsing=False)
+	tracking_key_tmp = args.get('show_title', None)
+
+	if tracking_key_tmp is None:
+		tracking_key = ''
+	else:
+		tracking_key = tracking_key_tmp[0];
+
+
+	url = url.replace(sys.argv[2][1:], '')
+
+	title = ''
+	show_title = ''
+
+	tvPlay(url, title, show_title, tracking_key)
+
+
 def tvPlay(url, title, show_title, tracking_key):
 	player = Player()
-	if title:
-		li = xbmcgui.ListItem(label=show_title + ' ' + title)
+
+	#if we start video without stopping the previous
+	if player.isPlaying(): 

```

Comment:
please do not use xbmc.Player(). use xbmcplugin.setResolvedUrl(handle, listitem, True) instead

Reply:
When creating the list, the provided urls  are not actual playable media urls. The media url  is given when the item has been selected and using xbmcplugin.setResolvedUrl creates a loop. Also I need the xbmc.Player() because of its events. Can you please direct me what exactly I should change?